### PR TITLE
Update old enum to remove pattern match warnings

### PR DESCRIPTION
**Pull request by RawToast**
There are more of these raising exhaustivity warnings during compilation. `Enumeration` as a type should ideally be avoided as it is not type safe

### CODE REVIEW

File: test/uk/gov/hmrc/advancevaluationrulings/models/RejectReasonSpec.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+      val formatted = Json.toJson(reason)
+      val result    = Json.fromJson[RejectReason](formatted)
+
+      result.get shouldBe reason
+    }
+
+  }
+
+}
```
Comment (amaalali) [inline test/uk/gov/hmrc/advancevaluationrulings/models/RejectReasonSpec.scala:72]: 👍🏽

File: app/uk/gov/hmrc/advancevaluationrulings/models/RejectReason.scala [hunk 2]
```diff
@@ -16,21 +16,16 @@
...
 package uk.gov.hmrc.advancevaluationrulings.models
 
-import play.api.libs.json.{Format, Json}
-import uk.gov.hmrc.advancevaluationrulings.models
+import enumeratum.{Enum, EnumEntry, PlayJsonEnum}
 
-object RejectReason extends Enumeration {
-  type RejectReason = Value
-  val APPLICATION_WITHDRAWN, ATAR_RULING_ALREADY_EXISTS, DUPLICATE_APPLICATION, NO_INFO_FROM_TRADER, OTHER = Value
+sealed abstract class RejectReason(override val entryName: String) extends EnumEntry
```
Comment (amaalali) [inline app/uk/gov/hmrc/advancevaluationrulings/models/RejectReason.scala:21]: I don't know if `entryName` semantically describes what this string represents for a `RejectReason`. Any thoughts on what a better name here could be?

**Review by amaalali — APPROVED** [review]
Looks good 👍🏽. Even though `RejectReason` is being used in areas where there is no test coverage, because the values aren't being used anywhere we can make this change now

File: app/uk/gov/hmrc/advancevaluationrulings/models/RejectReason.scala (hunk 2 above)
Comment (RawToast) [reply app/uk/gov/hmrc/advancevaluationrulings/models/RejectReason.scala:21]: I'd often just use `value` for these 😆

**Pull request merged**